app/main/views/agreements.py

```diff
@@ -10,13 +12,28 @@
 from ... import data_api_client
 
 
+status_labels = OrderedDict((
+    ("signed", "Waiting for countersigning"),
+    ("on-hold", "On hold"),
+    ("approved,countersigned", "Countersigned"),  # ugly key, but i don't want to start inventing new status values -
+                                                  # much easier to just act as a filter
+))
+
+
 @main.route('/agreements/<framework_slug>', methods=['GET'])
 @login_required
 @role_required('admin', 'admin-ccs-sourcing')
 def list_agreements(framework_slug):
     framework = data_api_client.get_framework(framework_slug)['frameworks']
+
+    status = request.args.get("status")
+    if status and status not in status_labels:
+        abort(400)
+
     supplier_frameworks = data_api_client.find_framework_suppliers(
-        framework_slug, agreement_returned=True
+        framework_slug,
+        agreement_returned=True,
```

Comment:
I'll fix the API to do the right thing, but it still wouldn't hurt to pass only one or the other in to the API client here.